decavision/model_testing/testing.py

```diff
@@ -253,3 +256,198 @@ def generate_classification_report(self, path):
+            if plot:
+                fig, ax = plt.subplots()
+                if self.model.name in ["Inception", "Xception"]:
+                    ax.imshow(image_tensor[0], interpolation='nearest')
+                else:
+                    ax.imshow(image_tensor[0].astype('uint8'), interpolation='nearest')
+                xlabel = 'Prediction :\n'
+                for (x, y) in zip(cls_pred_name, cls_pred_perc):
+                    xlabel += '{0}, {1:.2f}%\n'.format(x, y)
+                ax.set_xlabel(xlabel)
+                ax.set_xticks([])
+                ax.set_yticks([]) 
+                plt.tight_layout() 
+                plt.show()
+                fig.savefig("predicated_images/" + os.path.basename(image_path))                
+            else:
+                print('\nImage: ', image_path)
+                print("True label:", cls_true)
+                for i in range(len(cls_pred_perc)):
+                    print('Prediction: {} (probability {}%)'.format(cls_pred_name[i], round(cls_pred_perc[i])))
+
+    def evaluate(self, path):
+        """
+        Calculate the f1-score of the model on a dataset of images. The images must be
+        in single folder for each class.
+
+        Arguments:
+            path (str): location of the dataset
+        """
+        
+        generator = self._load_dataset(path)
+        results = self.model.evaluate(generator)
+        print('f1-score of', round(results[-1] * 100,3), '%')
+        
+    def generate_metrics(self, path, threshold=0.5):
+        """
+        Computes classification report and confusion matrix resulting from predictions on 
+        a dataset of images and prints the results. Images must be located in a single folder 
+        for each class. 
+        
+        Arguments:
+            path (str): location of the images
+            threshold (int): threshold for prediction (default to 0.5)
+        """
+    
+        # getting list of true and predicted labels
+        generator = self._load_dataset(path)
+        cls_true = generator.classes #true label for each image
+        labels = list(generator.class_indices.keys()) # list of total classes
+        cls_pred = self.model.predict(generator)
+        cls_pred = cls_pred > threshold
+        cls_true = [list(np.array(labels)[l]) for l in cls_true]
+        cls_pred_name = [list(np.array(labels)[l]) for l in cls_pred]
+        print('Labels & Predictions loaded for reports')
+
+        # binarizer
+        mlb = MultiLabelBinarizer(classes = labels)
+        y = mlb.fit_transform(cls_true) 
+        y_hat = mlb.fit_transform(cls_pred_name)
+
+        # classification report
+        print("\nClassification report")
+        print(classification_report(y, y_hat, target_names=labels, digits=4))
+
+        # confusion matrix
+        print("\n Confusion matrix")
+        print(multilabel_confusion_matrix(y, y_hat))
+            
+    def create_movie(self, image_path):
```

Comment:
If I understand correctly you need to run `classify_images` before running this to have the saved images? I think this method should be independent. So you should call the other method in it and have the path to images as input. Maybe keep the option to use premade images in case people have already called `classify_images`.